test/stress/main.go

```diff
@@ -446,11 +448,13 @@ Or you can terminate it any time by Ctrl + C.
 				lastEventTime := startTime
 				if len(esMap) > 0 && len(esTimeMap) > 0 {
 					timeout = o.idleTimeout
+					esLock.RLock()
 					for _, v := range esTimeMap {
 						if v.After(lastEventTime) {
 							lastEventTime = v
 						}
 					}
+					esLock.RUnlock()
 				}
 				if time.Since(lastEventTime).Seconds() > timeout.Seconds() {
 					fmt.Printf("Exited EventSource Pod %s due to no active events in the last %v\n", podName, o.idleTimeout)
```

Comment:
Looks like there already was a lock, so I just assume the crash occurred due to the read being unprotected.